engine/classes/Elgg/Assets/CssCompiler.php

```diff
@@ -56,31 +50,50 @@ public function compile($css, array $options = []) {
 		$config = (array) $this->config->css_compiler_options;
 
 		$options = array_merge($defaults, $config, $options);
-
-		$default_vars = array_merge($this->getCoreVars(), $this->getPluginVars());
-		$custom_vars = (array) elgg_extract('vars', $options, []);
-		$vars = array_merge($default_vars, $custom_vars);
-
-		$options['vars'] = $this->events->triggerResults('vars:compiler', 'css', $options, $vars);
+		$options['vars'] = $this->getCssVars($options);
 	
 		Crush::$process = new CssCrushProcess($options, ['type' => 'filter', 'data' => $css]);
 		return Crush::$process->compile()->__toString();
 	}
+	
+	/**
+	 * Fetches a combined set of CSS variables and their value
+	 *
+	 * @param array $compiler_options compiler arguments with potential custom variables
+	 * @param bool  $load_config_vars (internal) determines if config values are being loaded
+	 *
+	 * @return array
+	 */
+	public function getCssVars(array $compiler_options = [], bool $load_config_vars = true): array {
+		$default_vars = array_merge($this->getCoreVars(), $this->getPluginVars());
+		$custom_vars = (array) elgg_extract('vars', $compiler_options, []);
+		$vars = array_merge($default_vars, $custom_vars);
+		
+		$results = (array) $this->events->triggerResults('vars:compiler', 'css', $compiler_options, $vars);
+		
+		if (!$load_config_vars) {
+			return $results;
+		}
+		
+		return array_merge($results, (array) elgg_get_config('custom_theme_vars', []));
```

Comment:
Why not have the custom vars merged before the trigger?

Reply:
otherwise it is always part of the results, i need to get it without the custom theme vars in some cases. Also this way it is guaranteed always overriding the hook vars